doc/includes/RECIPES.md

```diff
@@ -778,3 +778,232 @@ explain select * from "Hero" where details#>>'{type}' = 'Hero';
+            to: "user_group_permission.user_id"
+          },
+          to: "user.id"
+        }
+      },
+      permissions: {
+        relation: Model.ManyToManyRelation,
+        modelClass: require("Permission"),
+        join: {
+          from: "group.id",
+          through: {
+            from: "user_group_permission.group_id",
+            extra: ["extra_attribute"],
+            to: "user_group_permission.permission_id"
+          },
+          to: "permission.id"
+        }
+      }
+    };
+  }
+}
+
+module.exports = Group;
+```
+
+```js
+// Permission.js
+const { Model } = require("objection");
+
+class Permission extends Model {
+  static get tableName() { return "permission"; }
+  static get relationMappings() {
+    return {
+      users: {
+        relation: Model.ManyToManyRelation,
+        modelClass: require("User"),
+        join: {
+          from: "permission.id",
+          through: {
+            from: "user_group_permission.permission_id",
+            extra: ["extra_attribute"],
+            to: "user_group_permission.user_id"
+          },
+          to: "user.id"
+        }
+      },
+      groups: {
+        relation: Model.ManyToManyRelation,
+        modelClass: require("Group"),
+        join: {
+          from: "permission.id",
+          through: {
+            from: "user_group_permission.permission_id",
+            extra: ["extra_attribute"],
+            to: "user_group_permission.group_id"
+          },
+          to: "group.id"
+        }
+      }
+    };
+  }
+}
+
+module.exports = Permission;
+```
+
+```js
+// UserGroupPermission.js
+const { Model } = require("objection");
+
+class UserGroupPermission extends Model {
+  static get tableName() { return "user_group_permission"; }
+  static get idColumn() { return ["user_id", "group_id", "permission_id"]; }
+  static get relationMappings() {
+    return {
+      user: {
+        relation: Model.BelongsToOneRelation,
+        modelClass: require("User"),
+        join: {
+          from: "user_group_permission.user_id",
+          extra: ["extra_attribute"],
+          to: "user.id"
+        }
+      },
+      group: {
+        relation: Model.BelongsToOneRelation,
+        modelClass: require("Group"),
+        join: {
+          from: "user_group_permission.group_id",
+          extra: ["extra_attribute"],
+          to: "group.id"
+        }
+      },
+      permission: {
+        relation: Model.BelongsToOneRelation,
+        modelClass: require("Permission"),
+        join: {
+          from: "user_group_permission.permission_id",
+          extra: ["extra_attribute"],
+          to: "permission.id"
+        }
+      }
+    };
+  }
+}
+
+module.exports = UserGroupPermission;
+```
+
+Here's how you can query your models:
+
+- `.*JoinRelation()`
+
+```js
+UserGroupPermission
+  .query()
+  .select(
+    "first_name",
+    "last_name",
+    "label",
+    "extra_attribute"
+  )
+  .joinRelation("user")
+  .joinRelation("permission")
```

Comment:
joinRelation now takes relation expressions. These two calls can be replaced with `joinRelation('[user, permission]')`